### ISSUE TITLE
MachOView needs to support lazy loading for content

### ISSUE THREAD

**Issue by rickmark** [Homebrew/ruby-macho#433]
As we start to parse contents of various regions, it would be helpful to not hold in memory `@raw_data` but materialize as needed in MachOView.

This can be accomplished by using the new file reference and seek / read to pull in the data lazy

**Comment by woodruffw** [comment]
Yeah, that's the idea behind #23. I opened that years ago and never got around to it, partially because the performance gains weren't substantial for Homebrew's use case. But I would happily accept a changeset for that now.

**Comment by rickmark** [comment]
Working towards it, curious how useful mmap files would be for these larger concerns but that is a native dependency for the OS so possible only as an optimization?

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by woodruffw** [comment]
Yeah, that's the trouble with `mmap`. We _could_ consider the `mmap` [gem](https://github.com/tenderlove/mmap) but in general we've tried to keep ruby-macho at zero dependencies because it gets vendored into Homebrew.
